### PR TITLE
Feat: devtron charts can be deployed now without gitops

### PR DESCRIPTION
**Pull request by vikramdevtron**
# Description

Devtron charts can be deploy with gitops (acd) and without gitops (helm).

Fixes # (issue): https://github.com/devtron-labs/devtron/issues/1653

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Devtron charts deployed without gitops 
- [x] Devtron charts deployed with gitops 
- [x] Handing db migration for existing apps.
- [x] App detail changes for resource tree for no gitops case tested.
- [x] App store - chart detail page status for both helm or acd type charts.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [x] I have tested it for all user roles
* [ ] I have added all the required unit/api test cases

### CODE REVIEW

**Review by nishant-d — APPROVED** [review]

File: scripts/sql/57_update_table.up.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE "public"."installed_apps" ADD COLUMN "deployment_app_type" varchar(50);
+
+update installed_apps set deployment_app_type='helm' WHERE app_id in (SELECT id from app WHERE app_offering_mode='EA_ONLY');
+
+update installed_apps set deployment_app_type='argo_cd' WHERE app_id in (SELECT id from app WHERE app_offering_mode='FULL');
```
Comment (manish-agrawal-ai) [inline scripts/sql/57_update_table.up.sql:5]: can you check if app_offering_mode is blank for some rows?

File: api/helm-app/HelmAppService.go [hunk 2]
```diff
@@ -407,7 +407,7 @@ func (impl *HelmAppServiceImpl) InstallRelease(ctx context.Context, clusterId in
 	}
 
 	installReleaseRequest.ReleaseIdentifier.ClusterConfig = config
-
+	impl.logger.Infow("helm install final request", "request", installReleaseRequest)
```
Comment (manish-agrawal-ai) [inline api/helm-app/HelmAppService.go:410]: Can we change it to debug?

File: scripts/sql/57_update_table.up.sql (hunk 1 above)
Comment (manish-agrawal-ai) [reply scripts/sql/57_update_table.up.sql:5]: != EA_ONLY

File: pkg/commonService/CommonService.go [hunk 3]
```diff
@@ -185,7 +185,7 @@ func (impl *CommonServiceImpl) GlobalChecklist() (*GlobalChecklist, error) {
 	}
 
 	if gitOps.Id > 0 {
```
Comment (manish-agrawal-ai) [inline pkg/commonService/CommonService.go:187]: you can remove the condition

File: pkg/appStore/deployment/tool/AppStoreDeploymentHelmService.go [hunk 4]
```diff
@@ -80,6 +80,7 @@ func (impl AppStoreDeploymentHelmServiceImpl) InstallApp(installAppVersionReques
 		return installAppVersionRequest, err
 	}
 
+	installAppVersionRequest.DeploymentAppType = util.PIPELINE_DEPLOYMENT_TYPE_HELM
```
Comment (manish-agrawal-ai) [inline pkg/appStore/deployment/tool/AppStoreDeploymentHelmService.go:83]: we can move it up

File: pkg/appStore/deployment/tool/gitops/AppStoreDeploymentArgoCdService.go [hunk 5]
```diff
@@ -66,22 +68,41 @@ func NewAppStoreDeploymentArgoCdServiceImpl(logger *zap.SugaredLogger, appStoreD
...
 }
 
 func (impl AppStoreDeploymentArgoCdServiceImpl) InstallApp(installAppVersionRequest *appStoreBean.InstallAppVersionDTO, ctx context.Context) (*appStoreBean.InstallAppVersionDTO, error) {
 	//step 2 git operation pull push
-	installAppVersionRequest, chartGitAttr, err := impl.appStoreDeploymentFullModeService.AppStoreDeployOperationGIT(installAppVersionRequest)
-	if err != nil {
-		impl.Logger.Errorw(" error", "err", err)
-		return installAppVersionRequest, err
+	isGitOpsConfigured := false
+	gitOpsConfig, err := impl.gitOpsRepository.GetGitOpsConfigActive()
```
Comment (manish-agrawal-ai) [inline pkg/appStore/deployment/tool/gitops/AppStoreDeploymentArgoCdService.go:78]: can we move this logic to upper layer?

Comment (vikramdevtron) [reply pkg/appStore/deployment/tool/gitops/AppStoreDeploymentArgoCdService.go]: done

File: pkg/appStore/deployment/tool/AppStoreDeploymentHelmService.go (hunk 4 above)
Comment (vikramdevtron) [reply pkg/appStore/deployment/tool/AppStoreDeploymentHelmService.go]: done

File: pkg/commonService/CommonService.go (hunk 3 above)
Comment (vikramdevtron) [reply pkg/commonService/CommonService.go]: done

File: api/helm-app/HelmAppService.go (hunk 2 above)
Comment (vikramdevtron) [reply api/helm-app/HelmAppService.go]: done

File: scripts/sql/57_update_table.up.sql (hunk 1 above)
Comment (vikramdevtron) [reply scripts/sql/57_update_table.up.sql]: done

**Review by manish-agrawal-ai — DISMISSED** [review]

**Review by manish-agrawal-ai — APPROVED** [review]

**Pull request merged**